canonical-web-and-design/maas-ui
# fix(ui): only allow hugepages and core pinning for LXD pods

## Done

- only allow hugepages and core pinning for LXD pods
- disable hugepage checkbox if none are free

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM details page of a virsh KVM
- Open the Compose form and check that the hugepages checkbox and pin cores radio button are disabled with tooltips
- Go to the KVM details page of a LXD KVM without any free hugepages (e.g. bolla)
- Check that the hugepages checkbox is disabled

## Fixes

Fixes #2456
